src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -242,29 +345,55 @@ public override void OnHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value,
         }
     }
 
-    private void ValidateHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
+    private void ValidateHeaderContent(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
```

Comment:
I know you call this ValidateHeaderContent in Http2Connection as well, but why is that? At first, I assumed it was only validating the header value, but it's mostly looking at the name (and rarely the value).

I wonder if we can share some of this logic between HTTP/2 and HTTP/3. I'm afraid someone in the future is going to fix some validation code in one place but not the other.

Reply:
The exceptions and status codes are different.